upgrade/util/util_test.go

```diff
@@ -0,0 +1,44 @@
+package util
+
+import (
+	"sync"
+	"testing"
+)
+
+func TestProgressMonitorInc(t *testing.T) {
+	expectedValue := 737
+	expectedCurrentProgressInPercentage := 74.0
+
+	pm := NewProgressMonitor("Test PM", 0, 1000)
+
+	var wg sync.WaitGroup
+
+	for i := 0; i < 3; i++ {
+		wg.Add(1)
+		go func(id int) {
+			defer wg.Done()
+			var count int
+			switch id {
+			case 0:
+				count = 250
+			case 1:
+				count = 247
+			case 2:
+				count = 240
+			}
+			for j := 0; j < count; j++ {
+				pm.Inc()
+			}
+		}(i)
+	}
+
+	wg.Wait()
+
+	currentValue, _, currentProgressInPercentage := pm.ReadCurrentProgress()
```

Comment:
nit: it's better to test for all public exported interfaces, but it's minor for this case because progress is pretty straightforward.

Reply:
Thank you! Fixed